Add Gauss_Kruger projection to registry (Transverse_Mercator alias)

jarobbins opened the pull request:
Closes #73

jarobbins committed:
Add Gauss_Kruger projection to registry (Transverse_Mercator alias)

Closes #73

FObermaier commented:
We should make sure that the `ProjectionsRegistry.CreateProjection` function does not override the passed class name, e.g.
```C#
internal static MathTransform CreateProjection(string className, IEnumerable<ProjectionParameter> parameters) {
    /*
     * ...
     */
    var res =  (MapProjection) Activator.CreateInstance(projectionType, parameters);
    if (!res.Name.Equals(className, StringComparison.InvariantCultureIgnoreCase))
    {
        res.Alias = res.Name;
        res.Name = className;
    }
    return res;
}
```

jarobbins committed:
Prevent override of passed class name in ProjectionsRegistry.CreateProjection

jarobbins commented:
@FObermaier Excellent point. Seems like that will help out with a few of the existing aliases, too.  I will add that to my branch.

Though, I do wonder if it wouldn't be better to create derived classes instead of just straight aliases.  Example:  `public class GaussKruger : TransverseMercator`.  Thoughts?

jarobbins committed:
Set Name property in three MapProjection classes

Some MapProjection-derived classes never had a name property set, so the ProjectionsRegistry.CreateProjection
method would throw NullRef on the res.Name to className comparison.

jarobbins commented:
@FObermaier Added logic to set the MapProjection.Name property to the passed-in className param (if not already equal).  Once that check was in, I saw there were some new failing tests.  A few projections never had their Name properties set, so I corrected that, as well.
